File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution(object):
    def searchRange(self, nums, target):
        def first():
            l=0
            r=len(nums)-1
            f=-1
            while l<=r:
                m=(l+r)//2

                if nums[m]==target:
                    f=m
                    r=m-1
                elif target>nums[m]:
                    l=m+1
                else:
                    r=m-1
                
            return f
        def last():
            l=0
            r=len(nums)-1
            la=-1
            while l<=r:
                m=(l+r)//2

                if nums[m]==target:
                    la=m
                    l=m+1

                elif target>nums[m]:
                    l=m+1
                else:
                    r=m-1
            return la
        return[first(),last()]
        

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int first=lb(nums,target);
        int last=ub(nums,target)-1;

        if(first==nums.length || nums[first]!=target){
            return new int[]{-1,-1};
        }
        return new int[]{first,last};
        
    }
    public int lb(int[] nums,int target){
        int low=0;
        int high=nums.length-1;

        int ans=nums.length;;
        while(low<=high){
            int mid=(low+high)/2;
            if(nums[mid]>=target){
                ans=mid;
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return ans;

    }
    public int ub(int[] nums,int target){
        int low=0;
        int high=nums.length-1;
        int ans=nums.length;
        while(low<=high){
            int mid=(low+high)/2;
            if(nums[mid]>target){
                ans=mid;
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return ans;
    }
}